### PR TITLE
fix: Add `--` to git diff command

### PR DESCRIPTION
**Pull request by bafulton**
## Problem

Some repositories contain a file in the root dir with the same name as that of the main branch. When this is the case, running a git diff against the main branch results in an error.

```
❯ ls
main
[...]

❯ git symbolic-ref HEAD --short
main

❯ git diff main
fatal: ambiguous argument 'main': both revision and filename
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```

## Solution

Append a `--` to the command as recommended by git. This clarifies we're providing the name of a _branch_ to diff, not the name of a file.

```
❯ git diff main --
```

Adding this parameter has no effect for repositories without a root file with the same name as the main branch, so we can safely add it to our command across the board.

### CODE REVIEW

File: cli/src/semgrep/git.py
Comment (bafulton) [inline cli/src/semgrep/git.py:175]: I've confirmed we can append other flags after this (eg, the `--merge-base` flag below. This param just needs to follow the branch name.

Comment (bafulton) [inline cli/src/semgrep/git.py]: I've confirmed we can append other flags after this (eg, the `--merge-base` flag below. This param just needs to follow the branch name param (`self._base_commit`).

**Comment by ievans** [comment]
@bafulton a test case would be great as well as I assume this code is / already moves to ocaml

**Review by aryx — DISMISSED** [review]
yep, what isaac said, please add an e2e tests, which will then lead you to also improve
the osemgrep code.

**Comment by aryx** [comment]
If modifying osemgrep and its OCaml code is too complicated, just add the e2e tests and we can push on your branch to fix osemgrep ourselves.

File: cli/tests/e2e/test_baseline.py
Comment (bafulton) [inline cli/tests/e2e/test_baseline.py:808]: I've confirmed this raises an exception on `develop`.

Comment (aryx) [reply cli/tests/e2e/test_baseline.py:808]: great

File: cli/tests/e2e/test_baseline.py
Comment (aryx) [inline cli/tests/e2e/test_baseline.py:800]: @ihji do you know why this test seems to pass also in osemgrep?
Ben didn't fix osemgrep code like pysemgrep but CI does not report any test failure ...
Does osemgrep and Git_wrapper.ml already handle correctly branch names that can be also files?

**Review by cgdolan — APPROVED** [review]
LGTM, thanks! Approving so we can get this change into pysemgrep ASAP since that's what ZCS is currently using and where we're seeing the most errors. Analysis team agreed to handle the osemgrep changes.

File: cli/tests/e2e/test_baseline.py
Comment (cgdolan) [reply cli/tests/e2e/test_baseline.py:800]: Closing this thread out to unblock the fix for pysemgrep. Let's continue the discussion for the osemgrep changes in a follow up.

**Pull request merged**
